tests/base/array_view_11.cc

```diff
@@ -0,0 +1,59 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// test for class ArrayView
+
+#include "../tests.h"
+
+#include <deal.II/base/array_view.h>
+
+
+void test ()
+{
+  {
+    std::vector<int> arr = {0,1,2,3,4,5,6,7,8,9};
+    ArrayView<int> view (arr);  // writable view
+    for (auto &el: view)
+      ++el;
+
+    int i=0;
+    for (auto &&it = arr.cbegin(); it != arr.cend(); ++it, ++i)
+      AssertThrow (*it == i+1, ExcInternalError());
+  }
+
+  {
+    std::vector<int> arr = {0,1,2,3,4,5,6,7,8,9};
+    const ArrayView<int> view (arr);  // writable view
+    for (auto &el: view)
+      ++el;
```

Comment:
I have some reservations about permitting this. I would rather have the `const` overload of `begin()` still return a `const_iterator`: I would prefer to not allow mutating things of type `const ArrayView<T>` in addition to `const ArrayView<const T>`.

Reply:
The point of this PR is to restore consistency with operator[].
```
  std::vector<int> arr = {0,1,2,3,4,5,6,7,8,9};
  const ArrayView<int> view (arr);  // writable view
  for (unsigned int i=1; i<view.size(); ++i)
    ++(arr[i]);
```
is equivalent to the above code and is already allowed.